pkg/controller/gcp.go

```diff
@@ -51,6 +51,7 @@ func (c *Controllers) SetupWithManager(mgr ctrl.Manager) error {
 		&container.GKEClusterClaimDefaultingController{},
 		&container.GKEClusterClaimController{},
```

Comment:
If it references a managed resource or claim directly then it will only pass the predicates of that controller (i.e. `v1beta1` or `v1alpha3`). If it is scheduled by label selectors then it is just like the race that happens against controllers from other stacks that also reconcile `KubernetesCluster` claims.

Reply:
This is not related to this PR but I am not sure whether we want `KubernetesCluster` to bind but not work as there is no nodepool created. We could somehow document it but users would still opt for the `beta` one instead of `alpha`.

I don't know, we really want something that solves this like https://github.com/crossplaneio/crossplane/pull/1118 ASAP at this point.